v4.1.0 (#519)
stellar/js-stellar-sdk: CHANGELOG.md

## Unreleased

## [v4.1.0](https://github.com/stellar/js-stellar-sdk/compare/v4.0.2...v4.1.0)

### Add

> - Add SEP0029 support. ([#516](https://github.com/stellar/js-stellar-sdk/issues/516))

Reviewer: ```suggestion
- Add SEP0029 (memo required) support. ([#516](https://github.com/stellar/js-stellar-sdk/issues/516))
```

  Extends `server.submitTransaction` to always run a memo required check before

>   sending the transaction, if any of the destinations requires a memo and the

Reviewer 2: ```suggestion
  sending the transaction.  If any of the destinations require a memo and the
```

>   sending the transaction, if any of the destinations requires a memo and the

Reviewer: ```suggestion
  sending the transaction. If any destination account requires a memo and the
```

  transaction doesn't include one, then an `AccountRequiresMemoError` will be thrown.  

  You can skip this check by passing `{skipMemoRequiredCheck: true}` to `server.submitTransaction`:

  ```
  server.submitTransaction(tx, {skipMemoRequiredCheck: true})
  ```

  The check runs for each operation of type:
   - `payment`
   - `pathPaymentStrictReceive`
   - `pathPaymentStrictSend`
   - `mergeAccount`

>   If the transaction includes a memo, then memo required checked is skipped.

Reviewer: ```
  If the transaction includes a memo, then memo required checking is skipped.
```
